src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx

```diff
@@ -228,16 +292,48 @@ export function PartneringFamilyScreen() {
           <Grid container spacing={0}>
             {partneringFamily.partneringFamilyInfo?.openReferral &&
               <Grid item xs={12}>
-                <h3 style={{ marginBottom: 0 }}>Arrangements</h3>
-                <Masonry columns={isDesktop ? isWideScreen ? 3 : 2 : 1} spacing={2}>
-                  {partneringFamily.partneringFamilyInfo?.openReferral?.arrangements?.slice()
-                  .sort((a,b) => sortArrangementsByStartDateDescThenCreateDateDesc(a,b))
-                  .map(arrangement => (
-                    <ArrangementCard key={arrangement.id}
-                      partneringFamily={partneringFamily} referralId={partneringFamily.partneringFamilyInfo!.openReferral!.id!}
-                      arrangement={arrangement} />
-                  )) || false}
-                  {permissions(Permission.CreateArrangement) && <Box sx={{textAlign: 'center'}}>
+                <div style={{ display: `flex`, justifyContent: `space-between`, maxWidth: `100%`, flexWrap: `wrap` }}>
+                  <div style={{ display: `flex`, justifyContent: `flex-start`, maxWidth: `100%`, flexWrap: `wrap` }}>
+                    <h3 style={{ margin: 0, display: `flex`, alignSelf: `center` }}>Arrangements</h3>
+                    <IconButton  
+                      aria-describedby={id}  
+                      onClick={handleFilterClick}
+                      size="small"
+                      color="primary" 
+                    >
+                      <FilterAlt />
+                    </IconButton>
+                    <Popover
+                      id={id}
+                      open={open}
+                      anchorEl={anchorEl}
+                      onClose={handleFilterClose}
+                      anchorOrigin={{
+                        vertical: 'bottom',
+                        horizontal: 'left',
+                      }}
+                    >
+                      <FormControl sx={{ m: 3 }} component="fieldset" variant="standard">
+                        <FormLabel component="legend">Filter Arrangements</FormLabel>
+                        <FormGroup>
+                          {
+                          arrangementFilterOptions.map((o) => {
+                            return (
```

Comment:
You actually don't need the `{ return ( /*...*/ ); }` -- you can just do something like:
```
<FormGroup>
  {arrangementFilterOptions.map(o => (
    <FormControlLabel /*...*/ />
  ))}
</FormGroup>
```
(and I'm even not sure if the inner parentheses are needed)

Reply:
Thanks for the suggestion - Fixed